components/browser/menu/src/main/java/mozilla/components/browser/menu/item/BrowserMenuItemToolbar.kt

```diff
@@ -62,13 +77,44 @@ class BrowserMenuItemToolbar(
      * @param iconTintColorResource Optional ID of color resource to tint the icon.
      * @param listener Callback to be invoked when the button is pressed.
      */
-    class Button(
+    open class Button(
         val imageResource: Int,
         val contentDescription: String,
         val iconTintColorResource: Int = 0,
         val listener: () -> Unit
     )
 
+    /**
+     * A button that either shows an primary state or an secondary state based on the provided
+     * <code>isInPrimaryState</code> lambda.
+     *
+     * @param primaryImageResource ID of a drawable resource to be shown as primary icon.
+     * @param primaryContentDescription The button's primary content description, used for accessibility support.
+     * @param primaryImageTintResource Optional ID of color resource to tint the primary icon.
+     * @param secondaryImageResource Optional ID of a different drawable resource to be shown as secondary icon.
+     * @param secondaryContentDescription Optional secondary content description for button, for accessibility support.
+     * @param secondaryImageTintResource Optional ID of secondary color resource to tint the icon.
+     * @param isInPrimaryState Lambda to return true/false to indicate if this button should be primary or secondary.
+     * @param disableInSecondaryState Optional boolean to disable the button when in secondary state.
+     * @param listener Callback to be invoked when the button is pressed.
+     */
+    open class TwoStateButton(
+        val primaryImageResource: Int,
+        val primaryContentDescription: String,
+        val primaryImageTintResource: Int = 0,
+        val secondaryImageResource: Int = primaryImageResource,
+        val secondaryContentDescription: String = primaryContentDescription,
+        val secondaryImageTintResource: Int = primaryImageTintResource,
+        val isInPrimaryState: () -> Boolean = { true },
```

Comment:
We could pass the current button's state in the listener below: `listener: (Boolean) -> Unit`